### PR TITLE
fix(auth): fixed test-local.sh in fxa-auth-server

### PR DESCRIPTION
**Pull request by dannycoates**
with hoisted eslint dependency `npm run lint` can't find
eslint on the path. We've got linting globally as a precommit
hook, it doesn't need to run on tests.

### CODE REVIEW

File: packages/fxa-auth-server/scripts/test-local.sh [hunk 1]
```diff
@@ -12,11 +12,10 @@ set -u
...
 
-npx ts-node ./scripts/gen_keys.js
-npx ts-node ./scripts/gen_vapid_keys.js
-npx ts-node ./scripts/oauth_gen_keys.js
-node ../fxa-auth-db-mysql/bin/db_patcher > /dev/null
-npm run lint
+node -r ts-node/register ./scripts/gen_keys.js
+node -r ts-node/register ./scripts/gen_vapid_keys.js
+node -r ts-node/register ./scripts/oauth_gen_keys.js
+node -r ts-node/register ../fxa-auth-db-mysql/bin/db_patcher > /dev/null
```
Comment (dannycoates) [inline packages/fxa-auth-server/scripts/test-local.sh:18]: register is faster than `npx ts-node`

File: packages/fxa-auth-server/test/local/routes/devices-and-sessions.js [hunk 2]
```diff
@@ -1604,110 +1604,110 @@ describe('/account/sessions', () => {
...
-          os: 'Windows',
-          userAgent: 'Firefox 50',
-          location: {
-            city: 'Toronto',
-            country: 'Canada',
-            state: 'Ontario',
-            stateCode: 'ON',
+      const noFormattedTime = ({ createdTimeFormatted, ...rest }) => rest;
+      assert.deepEqual(
+        result.map((x) => noFormattedTime(x)),
```
Comment (dannycoates) [inline packages/fxa-auth-server/test/local/routes/devices-and-sessions.js:1609]: this run was the last straw! no more failures on this bs

**Comment by 6a68** [comment]
How could I test this locally using the `circleci` CLI tool?

**Comment by dannycoates** [comment]
> How could I test this locally using the `circleci` CLI tool?

https://github.com/mozilla/fxa/blob/master/.circleci/README.md#local-testing

but do `test-auth-server`

**Comment by 6a68** [comment]
@dannycoates for some reason, fxa won't start on this branch or on master. Any thoughts?

```sh
[11:37:41 fxa (origin-fix-auth-test) ]$ yarn install
➤ YN0000: ┌ Resolution step
➤ YN0000: └ Completed in 0.8s
➤ YN0000: ┌ Fetch step
➤ YN0000: └ Completed in 1.57s
➤ YN0000: ┌ Link step
➤ YN0000: └ Completed in 15.8s
➤ YN0000: Done in 18.67s
[11:38:03 fxa (origin-fix-auth-test) ]$ npm start

> fxa@2.0.0 start /Users/jh/codez/github/fxa
> nps --prefix=start

nps is executing `start.default` : pm2 start _dev/pm2/infrastructure.config.js && _scripts/pm2-all.sh start && echo "Use 'npm stop' to stop all the servers"
[PM2] Spawning PM2 daemon with pm2_home=/Users/jh/.pm2
[PM2] PM2 Successfully daemonized
[PM2][WARN] Applications mysql, redis, memcache, sns, pubsub, firestore, sync, pushbox, email not running, starting...
[PM2] App [mysql] launched (1 instances)
[PM2] App [redis] launched (1 instances)
[PM2] App [memcache] launched (1 instances)
[PM2] App [sns] launched (1 instances)
[PM2] App [pubsub] launched (1 instances)
[PM2] App [firestore] launched (1 instances)
[PM2] App [sync] launched (1 instances)
[PM2] App [pushbox] launched (1 instances)
[PM2] App [email] launched (1 instances)
┌─────┬──────────────┬─────────────┬─────────┬─────────┬──────────┬────────┬──────┬───────────┬──────────┬──────────┬──────────┬──────────┐
│ id  │ name         │ namespace   │ version │ mode    │ pid      │ uptime │ ↺    │ status    │ cpu      │ mem      │ user     │ watching │
├─────┼──────────────┼─────────────┼─────────┼─────────┼──────────┼────────┼──────┼───────────┼──────────┼──────────┼──────────┼──────────┤
│ 8   │ email        │ default     │ 2.0.0   │ fork    │ 58196    │ 0s     │ 0    │ online    │ 0%       │ 1.4mb    │ jh       │ disabled │
│ 5   │ firestore    │ default     │ 2.0.0   │ fork    │ 58136    │ 0s     │ 0    │ online    │ 0%       │ 1.4mb    │ jh       │ disabled │
│ 2   │ memcache     │ default     │ 2.0.0   │ fork    │ 58108    │ 0s     │ 0    │ online    │ 0%       │ 1.4mb    │ jh       │ disabled │
│ 0   │ mysql        │ default     │ 2.0.0   │ fork    │ 58097    │ 0s     │ 0    │ online    │ 0%       │ 1.3mb    │ jh       │ disabled │
│ 4   │ pubsub       │ default     │ 2.0.0   │ fork    │ 58122    │ 0s     │ 0    │ online    │ 0%       │ 1.4mb    │ jh       │ disabled │
│ 7   │ pushbox      │ default     │ 2.0.0   │ fork    │ 58166    │ 0s     │ 0    │ online    │ 0%       │ 1.5mb    │ jh       │ disabled │
│ 1   │ redis        │ default     │ 2.0.0   │ fork    │ 58099    │ 0s     │ 0    │ online    │ 0%       │ 1.3mb    │ jh       │ disabled │
│ 3   │ sns          │ default     │ 2.0.0   │ fork    │ 58116    │ 0s     │ 0    │ online    │ 0%       │ 1.4mb    │ jh       │ disabled │
│ 6   │ sync         │ default     │ 2.0.0   │ fork    │ 58150    │ 0s     │ 0    │ online    │ 0%       │ 1.4mb    │ jh       │ disabled │
└─────┴──────────────┴─────────────┴─────────┴─────────┴──────────┴────────┴──────┴───────────┴──────────┴──────────┴──────────┴──────────┘
running start fxa services...
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! fxa-event-broker@1.172.2 start: `../../_scripts/check-url.sh localhost:9000/__heartbeat__ && pm2 start pm2.config.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the fxa-event-broker@1.172.2 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/jh/.npm/_logs/2020-06-01T18_39_26_689Z-debug.log
The script called "start.default" which runs "pm2 start _dev/pm2/infrastructure.config.js && _scripts/pm2-all.sh start && echo "Use 'npm stop' to stop all the servers"" failed with exit code 1 https://github.com/sezna/nps/blob/master/other/ERRORS_AND_WARNINGS.md#failed-with-exit-code
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! fxa@2.0.0 start: `nps --prefix=start`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the fxa@2.0.0 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/jh/.npm/_logs/2020-06-01T18_39_26_720Z-debug.log

**Comment by 6a68** [comment]
@dannycoates OK, I ran the CI tests locally as follows:

```sh
circleci config process .circleci/config.yml > .circleci/local.yml
circleci local execute -c .circleci/local.yml --job test-auth-server
```

6 of the tests failed. I don't know if the failures are meaningful or not, considering the same tests passed on CI for this branch. Thoughts?

```
(...snipped some code coverage noise...)

  2701 passing (9m)
  6 failing

  1) selectEmailServices with real redis:
       returned the correct results:
     Error: Timeout of 30000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves. (/home/circleci/project/packages/fxa-auth-server/test/local/senders/select_email_services.js)
      at listOnTimeout (internal/timers.js:549:17)
      at processTimers (internal/timers.js:492:7)

  2) lib/senders/sms:
       initialise:
         invalid data:
           wait a tick:
             called log.error correctly:

      AssertionError: expected 2 to equal 1
      + expected - actual

      -2
      +1
      
      at Context.<anonymous> (test/local/senders/sms.js:205:18)
      at processImmediate (internal/timers.js:456:21)
      at process.topLevelDomainCallback (domain.js:137:15)

  3) lib/senders/sms:
       initialise:
         unexpected response:
           wait a tick:
             called log.error correctly:

      AssertionError: expected 2 to equal 1
      + expected - actual

      -2
      +1
      
      at Context.<anonymous> (test/local/senders/sms.js:230:18)
      at processImmediate (internal/timers.js:456:21)
      at process.topLevelDomainCallback (domain.js:137:15)

  4) lib/senders/sms:
       "before each" hook for "failed correctly":
     Error: Timeout of 5000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves. (/home/circleci/project/packages/fxa-auth-server/test/local/senders/sms.js)
      at listOnTimeout (internal/timers.js:549:17)
      at processTimers (internal/timers.js:492:7)

  5) userAgent, mocked dependency
       "before each" hook for "recognises new mobile Sync library user agents on Android":
     Error: Timeout of 5000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves. (/home/circleci/project/packages/fxa-auth-server/test/local/user_agent.js)
      at processImmediate (internal/timers.js:456:21)
      at process.topLevelDomainCallback (domain.js:137:15)

  6) lib/verification-reminders:
       "before each" hook for "returned the correct result":
     Error: Timeout of 5000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves. (/home/circleci/project/packages/fxa-auth-server/test/local/verification-reminders.js)
      at processImmediate (internal/timers.js:456:21)
      at process.topLevelDomainCallback (domain.js:137:15)

**Comment by dannycoates** [comment]
> 6 of the tests failed. I don't know if the failures are meaningful or not, considering the same tests passed on CI for this branch. Thoughts?

These are some of the errors that can be triggered by running on too small of a docker host instance. It's a pretty good example of how resource intensive our code and testing infrastructure is. If you increase the "Resources" section in your (Docker for Mac?) preferences, you should be able to get a clean run.

**Comment by 6a68** [comment]
Ah, makes sense. Thanks for the explanation

**Pull request merged**